### DISCUSSION TITLE
Add new resource: Ec2 instance  - only auto reroll?

### DISCUSSION THREAD

**Discussion by followthebohns** [gravitational/teleport#43809] [Q&A]
I would like to know if the method of adding a new resource in teleport (AWS instance), as mentioned in the blog:

https://goteleport.com/blog/aws-ec2-instance-connect/

still works (or some way to reactivate it). Because I updated my self-hosted teleport from version 15.x to 16.0.3, and now the only way to add an ec2 instance is to just automate the process.

Thanks.

**Comment by marcoandredinis** [comment]
Hi,
We removed that option from the UI. We might remove it completely in the future.
The reason is that AWS added quota limits on its usage, which greatly reduces its usefulness.
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/eice-quotas.html

I would recommend that use the single instance flow which allows you to add a single instance to your cluster.
There's also this guide https://goteleport.com/docs/agents/join-services-to-your-cluster/aws-iam/ which explains how to easily onboard EC2 instances (and it doesn't require secret tokens).

Let me know if that works.
